src/core/CMakeLists.txt

```
@@ -246,6 +246,10 @@ find_package(ZXing REQUIRED)
 
 add_library(qfield_core STATIC ${QFIELD_CORE_SRCS} ${QFIELD_CORE_HDRS})
 
+string(SUBSTRING ${ZXing_VERSION} 0 1 ZXING_MAJOR_VERSION)
+target_compile_definitions(qfield_core
+                           PRIVATE ZXING_MAJOR_VERSION=${ZXING_MAJOR_VERSION})
```

```suggestion
target_compile_definitions(qfield_core
                           PRIVATE ZXING_MAJOR_VERSION=${ZXing_VERSION_MAJOR})
```

Up to you to decide if the variable should also be changed to `ZXING_VERSION_MAJOR` for consistency

---

@m-kuhn , you are right, feels better, I'll update now.